src/borg/archive.py

```diff
@@ -433,6 +433,7 @@ def write_checkpoint(self):
         self.save(self.checkpoint_name)
         del self.manifest.archives[self.checkpoint_name]
         self.cache.chunk_decref(self.id, self.stats)
```

Comment:
maybe adding a wait=True default to chunk_decref would be prettier.
so that one only has to do barriers if one actually requested an async delete.